File: searching/linear_search/python/linear_search.py
#!/usr/bin/python
"""
Script to perform linear search on an array
"""

def linear_search(array,x):
    """
	arguments:
	----------
		array: list
        x: int

	returns:
	--------
		i/str: int or str

	description:
	------------
    Function to return index of element in an array using linear search method.
    """
    #initializing index holder
    i = 0
    length = len(array)

    #iterate till array[i] matches with x while i is within the index span of
    #array length
    while i < length and array[i] != x:
        i += 1
    if i == len(array):
        return "Element not found in array"
    else:
        return i+1

if __name__ == "__main__":
    #getting array and element from user
    array = (raw_input("Enter elements of array with space as delimeter:\n").strip()).split(" ")
    array = [int(element) for element in array]
    element = int(raw_input("Enter the element you want to search:\n"))
    
    print linear_search(array,element)

File: searching/linear_search/c#/linear_search.cs
using System;

public class linear_search
{
	static void Main(string[] args)
	{
		int[] array = {13, 654, 48, 5, 186, 813, 81, 3};
		int index = -1, searchElement = 5;

		for (int i = 0; i < array.Length; i++)
		{
			if (array[i] == searchElement)
			{
				index = i;
				break;				
			}
		}
		if (index >= 0)
			Console.WriteLine(searchElement + " found at index " + index);
		else
			Console.WriteLine(searchElement + " not found in the given list");
	}	
}
